internal/logging/log.go

```diff
@@ -55,7 +55,7 @@ func (l Logger) WithName(name string) Logger {
 	return Logger{
 		Logger:        zapr.NewLogger(logger).WithName(name),
 		logging:       l.logging,
-		sugaredLogger: l.sugaredLogger,
+		sugaredLogger: logger.Sugar(),
```

Comment:
inited

https://github.com/envoyproxy/gateway/blob/4285f4e269ff309e81ec24b2f80fa9b5343f6a8f/internal/logging/log.go#L25-L44

Reply:
@qicz The issue with the current implementation `sugaredLogger: l.sugaredLogger` is that the logging level is incorrect. Please see the test `TestLoggerWithName` in this PR.

```
logger.Sugar().Debugf("debug message")
...
assert.Contains(t, capturedOutput, "debug message")
```

Reply:
misunderstand it